### PR TITLE
optimization w2grid.js

### PR DESCRIPTION
**Pull request by akaJes**
-removed unnecessary event handlers 
-DOM nodes looking more sanitized

### CODE REVIEW

File: src/w2grid.js [hunk 1]
```diff
@@ -4733,6 +4733,37 @@
...
+              $('#grid_'+ this.name +'_rec_' + w2utils.escapeId($(this).attr('recid'))).addClass('w2ui-record-hover')
+            })
+            .on('mouseout', 'tr', function(ev) {
+              $('#grid_'+ this.name +'_rec_' + w2utils.escapeId($(this).attr('recid'))).removeClass('w2ui-record-hover')
+            })
+            var self = this;
+            if(w2utils.isIOS)
+              records.add(frecords)
+              .on('click', 'tr', function(ev) {
+                self.dblClick($(this).attr('recid'), ev);
```
Comment (vitmalina) [inline src/w2grid.js]: this behavior is needed for mobile devices

File: src/w2grid.js [hunk 2]
```diff
@@ -4733,6 +4733,37 @@
 
             var gridBody = $('#grid_'+ this.name +'_body', obj.box).html(bodyHTML),
                 records = $('#grid_'+ this.name +'_records', obj.box);
+            var frecords = $('#grid_'+ this.name +'_frecords', obj.box);
+            records
+            .on('mouseover', 'tr', function(ev) {
+              $('#grid_'+ this.name +'_frec_' + w2utils.escapeId($(this).attr('recid'))).addClass('w2ui-record-hover')
```
Comment (vitmalina) [inline src/w2grid.js]: Should `this.name` be `self.name` and var self defined above it?

File: src/w2grid.js [hunk 3]
```diff
@@ -7235,20 +7252,6 @@
                     (isRowSelected && this.selectType == 'row' ? ' w2ui-selected' : '') +
                     (record.w2ui && record.w2ui.editable === false ? ' w2ui-no-edit' : '') +
                     (record.w2ui && record.w2ui.expanded === true ? ' w2ui-expanded' : '') + '" ' +
-                (summary !== true ?
```
Comment (vitmalina) [inline src/w2grid.js]: Did you account for summary records in your changed code?

File: src/w2grid.js [hunk 4]
```diff
@@ -7213,20 +7244,6 @@
...
-                    (w2utils.isIOS ?
-                        '    onclick  = "w2ui[\''+ this.name +'\'].dblClick(jQuery(this).attr(\'recid\'), event);"'
-                        :
-                        '    onclick  = "w2ui[\''+ this.name +'\'].click(jQuery(this).attr(\'recid\'), event);"'+
-                        '    oncontextmenu = "w2ui[\''+ this.name +'\'].contextMenu(jQuery(this).attr(\'recid\'), null, event);"'
-                     )
-                    : ''
-                ) +
-                (this.selectType == 'row' ?
```
Comment (vitmalina) [inline src/w2grid.js]: Did you account for selectType='cell' in your changed code?

File: src/w2grid.js [hunk 5]
```diff
@@ -4733,6 +4733,37 @@
...
+              })
+            else
+              records.add(frecords)
+              .on('click', 'tr', function(ev) {
+                self.click($(this).attr('recid'), ev);
+              })
+              .on('contextmenu', 'tr', function(ev) {
+                self.contextMenu($(this).attr('recid'), null, ev);
+              })
+
```
Comment (vitmalina) [inline src/w2grid.js]: Your code does look cleaner, but I am not sure it covers all use cases. Also, please note that due to lazy loading records might not be present, therefore, adding events with `on('click'...` might be wrong. I am not sure, though.

File: src/w2grid.js (hunk 2 above)
Comment (akaJes) [reply src/w2grid.js]: You are right, this hadn't been tested well

File: src/w2grid.js (hunk 3 above)
Comment (akaJes) [reply src/w2grid.js]: the summary recodrs parent don't have an event handler

File: src/w2grid.js (hunk 4 above)
Comment (akaJes) [reply src/w2grid.js]: fixed it in the code above

File: src/w2grid.js (hunk 5 above)
Comment (akaJes) [reply src/w2grid.js]: when parent node has an event handler - all children use it
[Event Handler Attachment](https://api.jquery.com/on/)

**Comment by vitmalina** [comment]
I think moving away from inline js is a good idea. Thank you for your contributeion.

**Pull request merged**